### PR TITLE
Centralized candidate import mailing to a single object

### PR DESCRIPTION
**Pull request by base10**
* Adds CandidateImportMailer
* Removes IcimsCandidateImportMailer and
  GreenhouseCandidateImportMailer and their respective templates
* Adjusts the candidate import controllers and candidate import models
* Provides a retry link for unsuccessful imports of candidates
  from iCIMS
* Greenhouse::CandidateName now has a "#name" method, which acts the
  same as "#to_s" does on that method, to allow for the refactoring
* For: https://trello.com/c/NGozb2N4/ and https://trello.com/c/Wr0dqCQe/
* Depends on #135 and #136 prior to merge

### CODE REVIEW

File: app/helpers/candidate_import_mailer_helper.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module CandidateImportMailerHelper
+  def retry_link
+    if @integration == "iCIMS"
```
Comment (jferris) [inline app/helpers/candidate_import_mailer_helper.rb]: This looks like integration-specific code in a generic helper. Some ideas to generalize it:

* Create a per-integration failure partial and render any integration-specific view code there.
* Create a `retriable?` method on the `Connection` subclasses so the generic code can decide whether or not the user is allowed to retry.

Comment (base10) [inline app/helpers/candidate_import_mailer_helper.rb]: Oh, smart. I like that.

Comment (base10) [inline app/helpers/candidate_import_mailer_helper.rb]: @jferris Fixed with 2a9e941. Coincidentally, a rereview would be appreciated.

File: app/mailers/candidate_import_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+class CandidateImportMailer < ApplicationMailer
+  def successful_import(candidate:, connection_type:, email:)
```
Comment (base10) [inline app/mailers/candidate_import_mailer.rb]: Tempted to change `connection_type` to `integration_id` for consistency

File: app/views/candidate_import_mailer/successful_import.en.text.erb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<%=
+  t(
+    "candidate_import_mailer.successful_import.message",
```
Comment (base10) [inline app/views/candidate_import_mailer/successful_import.en.text.erb]: Should be able to shorten this up to just `.message`

File: app/mailers/candidate_import_mailer.rb (hunk 2 above)
Comment (jferris) [inline app/mailers/candidate_import_mailer.rb]: Yeah, I think that would be an improvement.

File: app/services/greenhouse/candidates_importer.rb [hunk 4]
```diff
@@ -15,6 +15,7 @@ def initialize(mailer, connection_repo, params, secret_key, signature)
       @secret_key = secret_key
       @connection_repo = connection_repo
       @candidate_name = CandidateName.new(greenhouse_payload)
+      @integration_id = "greenhouse"
```
Comment (jferris) [inline app/services/greenhouse/candidates_importer.rb]: Can this be inlined?

File: app/services/icims/candidate_importer.rb [hunk 5]
```diff
@@ -6,6 +6,7 @@ def initialize(connection, mailer, params)
       @connection = connection
       @mailer = mailer
       @params = params
+      @integration_id = "icims"
```
Comment (jferris) [inline app/services/icims/candidate_importer.rb]: Can we inline this, too?

**Comment by jferris** [comment]
I had two minor comments. Looks good to merge after that.

File: app/services/greenhouse/candidates_importer.rb (hunk 4 above)
Comment (base10) [inline app/services/greenhouse/candidates_importer.rb]: I have multiple places in the class where I'm using this (see 13ae792), so my pref would be to keep an accessor for this.

File: app/services/icims/candidate_importer.rb (hunk 5 above)
Comment (base10) [inline app/services/icims/candidate_importer.rb]: I have multiple places in the class where I'm using this (see 13ae792), so my pref would be to keep an accessor for this.

File: app/services/greenhouse/candidates_importer.rb (hunk 4 above)
Comment (jferris) [inline app/services/greenhouse/candidates_importer.rb]: If you're worried about duplicating the value, what do you think about using a constant?

Comment (base10) [inline app/services/greenhouse/candidates_importer.rb]: I actually introduce one in #138 :D. I can do that here, though.

**Pull request merged**